src/server/generic_family.cc

```diff
@@ -524,9 +526,21 @@ OpResult<std::string> OpDump(const OpArgs& op_args, string_view key) {
   if (IsValid(it)) {
     DVLOG(1) << "Dump: key '" << key << "' successfully found, going to dump it";
     io::StringSink sink;
-    SerializerBase::DumpObject(it->second, &sink);
-    return sink.str();  // TODO: Add rvalue overload to str()
+    const PrimeValue& pv = it->second;
+
+    if (pv.IsExternal() && !pv.IsCool()) {
+      util::fb2::Future<string> future =
+          op_args.shard->tiered_storage()->Read(op_args.db_cntx.db_index, key, pv);
+
+      CompactObj co(future.Get());
+      SerializerBase::DumpObject(co, &sink);
+    } else {
+      SerializerBase::DumpObject(it->second, &sink);
```

Comment:
what if it is cool? do we support this inside dumpobject?
What is this cool thingy I dont remember this...

Reply:
it is very cool, actually. When we offload an object we may still keep it in-memory so if we hit it later, we can warm it up without hitting disk by moving it from the cool storage. Cool storage is just a FIFO queue of items and if we become low on memory we start discarding the oldest ones, making the values fully external.